src/main/java/seedu/lifeasier/storage/TaskStorage.java

```diff
@@ -0,0 +1,160 @@
+                case "deadline":
+                    rebuildDeadline(taskComponents, taskList, taskDescription, taskStatus);
+                    break;
+                case "event":
+                    rebuildEvent(taskComponents, taskList, taskDescription, taskStatus);
+                    break;
+                case "lesson":
+                    rebuildLesson(taskComponents, taskList, taskDescription, taskStatus);
+                    break;
+                default:
+                    System.out.println("Something went wrong while determining the tasks...");
+                    break;
+                }
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Encountered an error while reading from the save file - Data missing");
+        }
+    }
+
+    private void rebuildLesson(String[] taskComponents, ArrayList<Task> taskList, String description, Boolean status)
+            throws ArrayIndexOutOfBoundsException {
+        LocalDateTime lessonStartTime = fileCommand.convertToLocalDateTime(taskComponents[3]);
+        LocalDateTime lessonEndTime = fileCommand.convertToLocalDateTime(taskComponents[4]);
+
+        //Create new event in tasks
+        taskList.add(new Lesson(description, lessonStartTime, lessonEndTime, status));
+    }
+
+    private void rebuildEvent(String[] taskComponents, ArrayList<Task> taskList, String description, Boolean status)
+            throws ArrayIndexOutOfBoundsException {
+        LocalDateTime eventStartTime = fileCommand.convertToLocalDateTime(taskComponents[3]);
+        LocalDateTime eventEndTime = fileCommand.convertToLocalDateTime(taskComponents[4]);
+
+        //Create new event in tasks
+        taskList.add(new Event(description, eventStartTime, eventEndTime, status));
+    }
+
+    private void rebuildDeadline(String[] taskComponents, ArrayList<Task> taskList, String description, Boolean status)
+            throws ArrayIndexOutOfBoundsException {
+        LocalDateTime deadlineTimeInfo = fileCommand.convertToLocalDateTime(taskComponents[3]);
+
+        //Create new deadline in tasks
+        taskList.add(new Deadline(description, deadlineTimeInfo, status));
+    }
+
+    /**
+     * Writes information of TaskList onto the save file for storage whenever there is a change.
+     *
+     * @throws IOException When the file cannot be found or is corrupted.
+     */
+    public void writeToTaskSaveFile() {
+        try {
+            FileWriter fileWriter = new FileWriter(filePathTasks, true);
+            fileCommand.clearSaveFile(filePathTasks);
+
+            String dataToSave;
+            ArrayList<Task> taskList = tasks.getTaskList();
+            //Append each tasks information into save file for tasks
+            for (Task task : taskList) {
+                String taskType = task.getType();
+                switch (taskType) {
+                case "deadline":
+                    dataToSave = generateDeadlineSave(task, taskType);
```

Comment:
Same comment, you could consider changing the method name to better fit what it does.
```suggestion
                    dataToSave = convertDeadlineToString(task, taskType);
```

You could consider this for the methods below too.